pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -710,6 +712,48 @@ func (ctx *context) handleInboundResponse(response *Response) (stateAction,
 	}, connRecord, nil
 }
 
+// verifySignature verifies connection signature and returns connection
+func verifySignature(connSignature *ConnectionSignature) (*Connection, error) {
```

Comment:
what's the reason behind adding this function (hidden) here when there's a `Signer` interface, [found here](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/kms/api.go#L53), in the KMS (formerly wallet)? Anything sensitive in this function that shouldn't be exposed?

Reply:
Verify signature doesn't require private key. We can discuss weather it belongs to wallet.